### ISSUE TITLE
PPSSPP on ios: Insane battery drain while playing MUA2

### ISSUE THREAD

**Issue by Stevie-T** [hrydgard/ppsspp#14558]
### What should happen?
I am seeing huge battery drain while playing marvel ultimate alliance 2 with ppsspp on ios. I lose about 8-10% battery drain with 15 mins of gameplay. This does not seem to be the case with other games. Marvel ultimate alliance 1 loses around 3-4%, which seems acceptable. The game also overheats the phone. However, I reduced the heating when I reduced the emulated clock speed to 166hz. Can someone please investigate and improve the battery utilization for this game?

Hardware specifications -
iPhone xs max running ios 14.5. Battery health is at 94%.

PPSSPP settings -
build version - v1.11.3-634-ge7253ddf1
Backend - vulkan, Buffered rendering
Rendering resolution - 2x
CPU core - IR interpreter
Emulated clock - 166hz

**Comment by Gamemulatorer** [comment]
Because maybe using ir interpreter and not jit 🤔

**Comment by ANR2ME** [comment]
Does an older version of PPSSPP have better battery utilization?

**Comment by Stevie-T** [comment]
I have used older versions starting from last year but nothing much has changed for this game. I only saw slightly better performance once I reduced the emulated clock speed. Before that the I would lose around 1% battery drain per minute and experience insane overheating.

**Comment by unknownbrackets** [comment]
It's unfortunately expected that the IR interpeter is going to use more battery power than jit, and this may just be a demanding game.

We're always working to improve performance and reduce battery usage, but if this is mainly from Apple imposed limitations there may not be much we can do.

Do you know if Android has significant battery usage problems running the same game with jit?

You could also try posting some screenshots with "Show debug statistics" enabled during areas that seem to consume the most battery.  This displayed statistics may help point to specific things the game is doing, if there's something outside jit vs IR interpreter at work here.

-[Unknown]

**Comment by hrydgard** [comment]
Yeah, in addition to CPU usage, it might also be the case that the game does something very inefficient on the GPU. In that case we'd expect to see similar behavior on Android.

**Comment by Stevie-T** [comment]
As requested, I have captured 3 min gameplay with 'show debug statistics' switched on. This run has PPSSPP set with custom settings mentioned above. We lose 3% battery life after 3 mins of gameplay.

[https://drive.google.com/file/d/1tylSzaaaUYv7KQeN86pwALfS982xJF_e/view?usp=sharing](https://drive.google.com/file/d/1tylSzaaaUYv7KQeN86pwALfS982xJF_e/view?usp=sharing)

I also recorded the same gameplay with PPSSPP default settings. Here we see don't see any battery drain. However, iPhone battery indicator does not show consistent drain when near 100% charge.

[https://drive.google.com/file/d/1n-wCgnafhGVoo-bBiLNvPY9T-mjPpJan/view?usp=sharing](https://drive.google.com/file/d/1n-wCgnafhGVoo-bBiLNvPY9T-mjPpJan/view?usp=sharing)

**Comment by unknownbrackets** [comment]
This is interesting:
![Slowest syscall: sceAudioOutput2OutputBlocking at 12.57ms in a single frame](https://user-images.githubusercontent.com/191233/128617484-92884bfe-c98e-4c3d-a6a3-0417f2a77593.png)

There's also this:
![Sudden drop in FPS and spike in syscall time](https://user-images.githubusercontent.com/191233/128617542-14a1f602-b97f-4e90-a43f-d0a924257465.png)

And this:
![Slowest syscall: sceKernelGetSystemTimeWide at 20.89ms](https://user-images.githubusercontent.com/191233/128617563-93f013c9-6d50-420d-bc66-7b8b7097ef27.png)

These are in the "custom settings" video, which I think you're saying has expected battery drain.  The second noted event seems to be an external slowdown (it doesn't seem like it's from PPSSPP itself, but maybe from a background app - could be power/heat management on the device.)  The other two look like potential performance issues either in the game or in PPSSPP.

For the default settings, we a similar pattern.  It also had an apparent external slowdown event.  It generally looked similar, but typical usage seemed higher (although we're still talking a few milliseconds out of the entire frame.)  It's possible that at this speed, it's rendering internally more frames, correctly or incorrectly.

`sceAudioOutput2OutputBlocking` is interesting because it already eats a decent chunk of cycles, and I wonder if it's calling it in an error condition in a loop (which would currently perform potentially badly.)

`sceKernelGetSystemTimeWide` already has correct timing, so it's strange that spikes, and may likely have happened on a real PSP.  That makes me think a game patch/cheat could be written to improve performance a bit, at least in those areas.

-[Unknown]

**Comment by Gamemulatorer** [comment]
I have experience this battery drain is fast playing ppsspp for an hours using the latest ppsspp development version for android or maybe because my android 11 update? 😅
![Screenshot_2021-08-16-22-02-47-207_com miui securitycenter](https://user-images.githubusercontent.com/37603562/129576714-f0e38a6f-4e6d-4579-aa20-c3b978962264.jpg)
